### PR TITLE
CIV-2200 DeadLine Calculator to send app to other party

### PR DESCRIPTION
**Pull request by karthick-mohanasundaram-HMCTS**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-2200

### Related PR ###

https://github.com/hmcts/civil-general-apps-ccd-definition/pull/171

### Change description ###

Judge - Make Decision Send application to other party and request hearing details

System will wait for a response for 5 working days (not including weekends or bank holidays). To calculate the end date for the response – add a week to the submitted date – so usually it’s 4pm a week later but if any UK government bank holidays fall in between this a day is added for each bank holiday – and again its 4pm on that day.

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/JudicialDecisionHandler.java [hunk 1]
```diff
@@ -358,13 +363,32 @@ private CallbackResponse gaValidateMakeAnOrder(CallbackParams callbackParams) {
...
+
         GAJudicialRequestMoreInfo judicialRequestMoreInfo = caseData.getJudicialDecisionRequestMoreInfo();
         List<String> errors = judicialRequestMoreInfo != null
             ? validateDatesForRequestMoreInfoScreen(judicialRequestMoreInfo)
             : Collections.emptyList();
 
+        if (judicialRequestMoreInfo != null
+            && SEND_APP_TO_OTHER_PARTY.equals(judicialRequestMoreInfo.getRequestMoreInfoOption())) {
+            String endDateForMoreInfoSubmission = deadlinesCalculator
+                .calculateApplicantResponseDeadline(
```
Comment (smitatalawadekar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/JudicialDecisionHandler.java:376]: We should not be converting date to String.... The current deadline is being saved as date but we have a ticket open to change it to Date... In similar way, all the dates including this deadline should be LocalDateTime.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/JudicialDecisionHandler.java [hunk 2]
```diff
@@ -418,11 +442,14 @@ private SubmittedCallbackResponse buildConfirmation(CallbackParams callbackParam
                         throw new IllegalArgumentException("Missing data during submission of judicial decision");
                     }
                 } else if (SEND_APP_TO_OTHER_PARTY.equals(requestMoreInfo.getRequestMoreInfoOption())) {
+                    LocalDateTime submissionEndDate = LocalDateTime.parse(caseData
```
Comment (smitatalawadekar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/JudicialDecisionHandler.java:445]: string parsing to date would have been redundant if it was a date as mentioned in previous comment.

File: src/main/java/uk/gov/hmcts/reform/civil/service/GeneralAppsDeadlinesCalculator.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package uk.gov.hmcts.reform.civil.service;
+
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+
+@Service
+public class GeneralAppsDeadlinesCalculator extends DeadlinesCalculator {
```
Comment (smitatalawadekar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/service/GeneralAppsDeadlinesCalculator.java:8]: Do we need 2 separate classes? In civil-service, there are 2 classes as the default calculator (created for claims) does not produce date as per our requirements so we had to extend it to override a method for our purpose. We can have a single calculator here

Comment (karthick-mohanasundaram-HMCTS) [reply src/main/java/uk/gov/hmcts/reform/civil/service/GeneralAppsDeadlinesCalculator.java]: Done. Cheers!

**Review by smitatalawadekar-hmcts — APPROVED** [review]

**Review by vasudevganesanhmcts — APPROVED** [review]

**Review by vasudevganesanhmcts — APPROVED** [review]

**Pull request merged**